hydroflow_lang/src/graph/ops/join_fused.rs

```diff
@@ -0,0 +1,333 @@
+use proc_macro2::{Ident, TokenStream};
+use quote::{quote_spanned, ToTokens};
+use syn::spanned::Spanned;
+use syn::{parse_quote, Expr, ExprCall};
+
+use super::{
+    DelayType, FlowProperties, FlowPropertyVal, OperatorCategory, OperatorConstraints,
+    OperatorWriteOutput, Persistence, WriteContextArgs, RANGE_0, RANGE_1,
+};
+use crate::diagnostic::{Diagnostic, Level};
+use crate::graph::{OpInstGenerics, OperatorInstance};
+
+/// > 2 input streams of type <(K, V1)> and <(K, V2)>, 1 output stream of type <(K, (V1, V2))>
+///
+/// `join_fused` takes two arguments, they are the left hand side and right hand side reducing/folding operation respectively.
+/// The syntax is a bit unique for them, but it is similar to specifying an enum in rust. For example:
+/// ```hydroflow,ignore
+/// join_fused(Fold(|| "default value", |x, y| FoldingFunction(x, y)), Reduce(|x, y| ReducingFunction(x, y)))
+/// join_fused(FoldFrom(|x| "conversion function", |x, y| FoldingFunction(x, y)), Reduce(|x, y| ReducingFunction(x, y)))
+/// ```
+/// The three currently supported fused operator types are `Fold(Fn() -> A, Fn(A, T) -> A)`, `Reduce(Fn(A, A) -> A)`, and `FoldFrom(Fn(T) -> A, Fn(A, T) -> A)`
```

Comment:
Please add an example of `FoldFrom` usage to the doc.

Reply:
@jhellerstein if you have any neat canonical examples of such a structure that requires this thing. I'd like to include it, my current example is very lame and contrived.